### ISSUE TITLE
MissingPluginException(No implementation found for method getApplicationDocumentsDirectory on channel plugins.flutter.io/path_provider)

### ISSUE THREAD

**Issue by the-coding-turtle** [hivedb/hive#881]
Hi,

I am getting the following Exception when running my tests.
`MissingPluginException(No implementation found for method getApplicationDocumentsDirectory on channel plugins.flutter.io/path_provider)`

I have used the library before in similar ways without any problems. 
But I cannot figure out the reason for the exception now...

**Steps to Reproduce**
Created a new Flutter-Project and received this error. See repo below.

**Code sample**
Here is the Repo:
[https://github.com/the-coding-turtle/cam-watcher](https://github.com/the-coding-turtle/cam-watcher)

Here is the Class:
[https://github.com/the-coding-turtle/cam-watcher/blob/main/lib/src/services/hive_service.dart](https://github.com/the-coding-turtle/cam-watcher/blob/main/lib/src/services/hive_service.dart)

Here is the Test:
[https://github.com/the-coding-turtle/cam-watcher/blob/main/test/src/services/hive_service_test.dart](https://github.com/the-coding-turtle/cam-watcher/blob/main/test/src/services/hive_service_test.dart)

Here the latest GithubAction with the error:
[https://github.com/the-coding-turtle/cam-watcher/runs/4792362577?check_suite_focus=true](https://github.com/the-coding-turtle/cam-watcher/runs/4792362577?check_suite_focus=true)

**Version**
 - Platform: Android, Windows
 - Flutter version: 2.5.3
 - Hive version: 2.0.5

**Comment by jpirovano** [comment]
same issue here

**Comment by BlackCatDev-IO** [comment]
I believe this is a path provider issue and not a Hive issue. See [this  answer on SO ](https://stackoverflow.com/questions/62597011/mock-getexternalstoragedirectory-on-flutter)and  [this full example](https://github.com/flutter/plugins/blob/master/packages/path_provider/path_provider/test/path_provider_test.dart) for more info. Neither of which are my code, but when I ran into this same error (which occurs for me using Hive or GetStorage), mocking a PathProviderPlatform in my tests fixed it for me. 

In short, you can create a `test_utils` file that has this mock. 

```
const String kTemporaryPath = 'temporaryPath';
const String kApplicationSupportPath = 'applicationSupportPath';
const String kDownloadsPath = 'downloadsPath';
const String kLibraryPath = 'libraryPath';
const String kApplicationDocumentsPath = 'applicationDocumentsPath';
const String kExternalCachePath = 'externalCachePath';
const String kExternalStoragePath = 'externalStoragePath';

class FakePathProviderPlatform extends Fake
    with MockPlatformInterfaceMixin
    implements PathProviderPlatform {
  @override
  Future<String?> getTemporaryPath() async {
    return kTemporaryPath;
  }

  @override
  Future<String?> getApplicationSupportPath() async {
    return kApplicationSupportPath;
  }

  @override
  Future<String?> getLibraryPath() async {
    return kLibraryPath;
  }

  @override
  Future<String?> getApplicationDocumentsPath() async {
    return kApplicationDocumentsPath;
  }

  @override
  Future<String?> getExternalStoragePath() async {
    return kExternalStoragePath;
  }

  @override
  Future<List<String>?> getExternalCachePaths() async {
    return <String>[kExternalCachePath];
  }

  @override
  Future<List<String>?> getExternalStoragePaths({
    StorageDirectory? type,
  }) async {
    return <String>[kExternalStoragePath];
  }

  @override
  Future<String?> getDownloadsPath() async {
    return kDownloadsPath;
  }
}
```

Then in your test `setUpAll` you can add this line. 

```
    PathProviderPlatform.instance = FakePathProviderPlatform();
```

**Comment by Rodsevich** [comment]
For what I've been reading, the path_provider dependency was fixed on its 3.1.0 version. We need to change that in our pubspec.yaml.

**Comment by the-coding-turtle** [comment]
@BlackCatDev-IO Thank you very much! It works now.
